pennylane/_qubit_device.py

```diff
@@ -543,13 +574,18 @@ def analytic_probability(self, wires=None):
         """
         raise NotImplementedError
 
-    def estimate_probability(self, wires=None):
+    def estimate_probability(self, wires=None, shot_range=None, bin_size=None):
         """Return the estimated probability of each computational basis state
         using the generated samples.
 
         Args:
             wires (Iterable[Number, str], Number, str, Wires): wires to calculate
                 marginal probabilities for. Wires not provided are traced out of the system.
+            shot_range (tuple[int]): 2-tuple of integers specifying the range of samples
+                to use. If not specified, all samples are used.
+            bin_size (int): Divides the shot range into bins of size ``bin_size``, and
+                returns the measurement statistic separately over each bin. If not
+                provided, the entire shot range is treated as a single bin.
```

Comment:
I am slow on the uptake here...what is the role of shot_range vs bin_size and how do they relate to the shots_vector?

Reply:
Ah, so say you have `shots=[5, (10, 3), 100]`.

For the second element (10 shots, 3 copies) we will want to use samples number 5 to 35, performing the expectation value in bins of size 10.

So in this case `shot_range=[5, 35]`, `bin_size=10`.

Maybe there are better argument names I could use?

Reply:
Aah! no, that makes sense.